Regression on v0.38

### Checks

- [X] I have checked that this issue has not already been reported.
- [X] I have confirmed this bug exists on the [latest version](https://crates.io/crates/polars) of Polars.

### Reproducible example

With this declaration in `Cargo.toml`:
```toml
[dependencies.polars]
version = "0.38.0"
default-features = false
features = [
    "bigidx",
    "cse",
    "fmt",
    "fmt_no_tty",
    "ndarray",
    "parquet",
    "performant",
    "temporal",

    "rolling_window",
    "date_offset",

    "abs",
    "arg_where",
    "checked_arithmetic",
    "chunked_ids",
    "coalesce",
    "cum_agg",
    "cumulative_eval",
    "dataframe_arithmetic",
    "dot_product",
    "group_by_list",
    "lazy",
    "list_any_all",
    "list_drop_nulls",
    "list_eval",
    "list_gather",
    "list_sample",
    "list_sets",
    "meta",
    "mode",
    "partition_by",
    "pct_change",
    "peaks",
    "pivot",
    "product",
    "range",
    "rank",
    "reinterpret",
    "repeat_by",
    "sign",
    "true_div",
]
```

### Log output

```shell
On Windows and Linux, when i build i have these errors:

error[E0599]: no variant or associated item named `Categorical` found for enum `polars_core::datatypes::DataType` in the current scope
   --> ~\.cargo\registry\src\index.crates.io-6f17d22bba15001f\polars-ops-0.38.1\src\chunked_array\list\sets.rs:424:23
    |
424 |     if let (DataType::Categorical(_, _), DataType::Categorical(_, _)) =
    |                       ^^^^^^^^^^^ variant or associated item not found in `DataType`

error[E0599]: no variant or associated item named `Categorical` found for enum `polars_core::datatypes::DataType` in the current scope
   --> ~\.cargo\registry\src\index.crates.io-6f17d22bba15001f\polars-ops-0.38.1\src\chunked_array\list\sets.rs:424:52
    |
424 |     if let (DataType::Categorical(_, _), DataType::Categorical(_, _)) =
    |                                                    ^^^^^^^^^^^ variant or associated item not found in `DataType`

error[E0425]: cannot find function `make_list_categoricals_compatible` in this scope
   --> ~\.cargo\registry\src\index.crates.io-6f17d22bba15001f\polars-ops-0.38.1\src\chunked_array\list\sets.rs:427:18
    |
427 |         (a, b) = make_list_categoricals_compatible(a, b)?;
    |                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ not found in this scope

error[E0308]: mismatched types
  --> ~\.cargo\registry\src\index.crates.io-6f17d22bba15001f\polars-ops-0.38.1\src\frame\join\hash_join\sort_merge.rs:35:27
   |
35 |     (flatten_par(&lefts), flatten_par(&rights))
   |                           ^^^^^^^^^^^^^^^^^^^^ expected `Vec<Option<u64>>`, found `Vec<NullableIdxSize>`
   |
   = note: expected struct `Vec<Option<u64>>`
              found struct `Vec<NullableIdxSize>`
```


### Issue description

This cannot build with version 0.38, but that build with version 0.37; the only change is just the version in `Cargo.toml`.

### Expected behavior

This should build with version 0.38.

### Installed versions

<details>
Rust 1.76

</details>

--- a/crates/polars-ops/src/chunked_array/list/sets.rs
+++ b/crates/polars-ops/src/chunked_array/list/sets.rs
@@ -421,6 +421,7 @@ pub fn list_set_operation(
     b.prune_empty_chunks();
 
     // Make categoricals compatible
+    #[cfg(feature = "dtype-categorical")]
     if let (DataType::Categorical(_, _), DataType::Categorical(_, _)) =
         (&a.inner_dtype(), &b.inner_dtype())
     {

--- a/crates/polars-pipe/src/executors/sinks/memory.rs
+++ b/crates/polars-pipe/src/executors/sinks/memory.rs
@@ -41,7 +41,7 @@ impl MemTracker {
     }
 
     /// This shouldn't be called often as this is expensive.
-    fn refresh_memory(&self) {
+    pub fn refresh_memory(&self) {
         self.available_mem
             .store(MEMINFO.free() as usize, Ordering::Relaxed);
     }
@@ -54,6 +54,12 @@ impl MemTracker {
         if fetch_count % (self.refresh_interval * self.thread_count) == 0 {
             self.refresh_memory()
         }
+        self.available_mem.load(Ordering::Relaxed)
+    }
+
+    pub(super) fn get_available_latest(&self) -> usize {
+        self.refresh_memory();
+        self.fetch_count.store(0, Ordering::Relaxed);
         self.available_mem.load(Ordering::Relaxed)
     }
 

--- a/crates/polars-pipe/src/executors/sinks/sort/ooc.rs
+++ b/crates/polars-pipe/src/executors/sinks/sort/ooc.rs
@@ -117,7 +117,10 @@ pub(super) fn sort_ooc(
 ) -> PolarsResult<FinalizedSink> {
     let samples = samples.to_physical_repr().into_owned();
     // Try to use available memory. At least 32MB per spill.
-    let spill_size = std::cmp::max(memtrack.get_available() / (samples.len() * 2), 1 << 25) as u64;
+    let spill_size = std::cmp::max(
+        memtrack.get_available_latest() / (samples.len() * 3),
+        1 << 25,
+    ) as u64;
 
     // we collect as I am not sure that if we write to the same directory the
     // iterator will read those also.

